### PR TITLE
Att consent lets try that again

### PR DESCRIPTION
**Pull request by taquitos**
Part 3 of 3 
Breaking apart my previous PR (https://github.com/RevenueCat/purchases-ios/pull/1400)
This time the tests pass..... on my machine.
confetti

This is for [CF-281]

[CF-281]: https://revenuecats.atlassian.net/browse/CF-281?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by taquitos** [comment]
Note, the only thing you should review here is the use of the attributionFetcher in `SubscriptionAPI` the rest are changes that I merged into it from the other branches. I'll rebase on those changes once they are merged.

File: Purchases/Attribution/AttributionFetcher.swift
Comment (taquitos) [inline Purchases/Attribution/AttributionFetcher.swift:99]: This is new

File: Purchases/Attribution/TrackingManagerProxy.swift
Comment (taquitos) [inline Purchases/Attribution/TrackingManagerProxy.swift:26]: This is new

File: Purchases/Networking/SubscribersAPI.swift
Comment (taquitos) [inline Purchases/Networking/SubscribersAPI.swift:95]: This is new

File: Tests/UnitTests/Networking/Backend/__Snapshots__/BackendPostReceiptDataTests/testCachesRequestsForSameReceipt.1.json
Comment (taquitos) [inline Tests/UnitTests/Networking/Backend/__Snapshots__/BackendPostReceiptDataTests/testCachesRequestsForSameReceipt.1.json:13]: All these attributes are new in the test fixtures.

File: Purchases/Networking/SubscribersAPI.swift
Comment (aboedo) [inline Purchases/Networking/SubscribersAPI.swift:94]: nitpick: we have a [var called key](https://github.com/RevenueCat/purchases-ios/blob/main/Purchases/SubscriberAttributes/ReservedSubscriberAttributes.swift#L20) that's an alias for `rawValue` for readability

```suggestion
        subscriberAttributesByKey[ReservedSubscriberAttribute.consentStatus.key] = consentStatus
```

**Review by aboedo — APPROVED** [review]

File: Purchases/Networking/SubscribersAPI.swift
Comment (taquitos) [reply Purchases/Networking/SubscribersAPI.swift]: ✅

**Comment by taquitos** [comment]
> @taquitos I updated the base to #1406 so the diff from that branch doesn't show up here too, feel free to undo if you plan to merge this one and close the other one

Thanks! I just merged the other ones and then rebased on `main` so the changed files list now shows reality 😄

**Pull request merged**
